src/Bounds.cpp

```diff
@@ -1235,8 +1235,26 @@ class Bounds : public IRVisitor {
                                        !b_interval.min.type().is_uint() &&
                                        can_prove(b_interval.min < 0 &&
                                                  b_interval.min > -t.bits())) {
-                                interval.min = a_interval.min >> abs(b_interval.min);
+                                if (a_interval.min.type().is_uint() || can_prove(a_interval.min >= 0)) {
```

Comment:
Is it a significant win in terms of execution time to check for is_uint() here? (I'd hope that can_prove(e>=0) would always catch these cases since all uints are >=0)

Reply:
I imagine it would be at least decent - `can_prove` calls `remove_likelies`, `common_subexpression_elimination` and `simplify` which could be quite expensive if `a_interval.min` is a complicated expression, but `is_uint()` should be extremely fast.

Reply:
in that case, should you also add it on line 1254? 

`a_interval.min.type.is_uint() || can_prove(a_interval.min >= 0)`

Reply:
(I also now idly wonder if it would be worth having can_prove check for the expr being `foo >= 0` or `foo < 0` and doing an early-out if the LHS is uint...)

Reply:
> in that case, should you also add it on line 1254?

sure thing. great catch, thanks!

Reply:
Fixed in 3055665